drivers/net/ethernet/renesas/rswitch.c

```diff
@@ -3982,6 +4051,65 @@ static struct notifier_block vlan_notifier_block __read_mostly = {
 	.notifier_call = vlan_device_event,
 };
 
+static void rswitch_netevent_work(struct work_struct *work)
+{
+	struct rswitch_net *rn;
+	struct rswitch_private *priv;
+	struct rswitch_device *rdev;
+	struct rswitch_ipv4_route *routing_list;
+	struct l3_ipv4_fwd_param_list *l3_param_list;
+
+	rn = net_generic(&init_net, rswitch_net_id);
+	priv = rn->priv;
+
+	mutex_lock(&priv->rdev_list_lock);
+	mutex_lock(&priv->ipv4_forward_lock);
+	list_for_each_entry(rdev, &priv->rdev_list, list)
+		list_for_each_entry(routing_list, &rdev->routing_list, list)
+			list_for_each_entry(l3_param_list, &routing_list->param_list, list) {
+				/* Skip params related to dst interface route (zero src) */
+				if (l3_param_list->param->src_ip)
+					rswitch_modify_l3fwd(l3_param_list->param,
+							     !priv->ipv4_forward_enabled);
+			}
+	mutex_unlock(&priv->ipv4_forward_lock);
+	mutex_unlock(&priv->rdev_list_lock);
+
+	kfree(work);
+}
+
+static int rswitch_netevent_cb(struct notifier_block *unused, unsigned long event, void *ptr)
+{
+	struct net *net = ptr;
+	struct rswitch_net *rn;
+	struct rswitch_private *priv;
+	struct work_struct *work;
+
+	if (event != NETEVENT_IPV4_FORWARD_UPDATE)
+		return 0;
```

Comment:
NIT: it can be changed to NOTIFY_DONE macro (it's equal to zero, so it's just a cosmetic change)